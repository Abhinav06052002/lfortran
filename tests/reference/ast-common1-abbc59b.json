{
    "basename": "ast-common1-abbc59b",
    "cmd": "lfortran --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/common1.f90",
    "infile_hash": "d7a7a1f2f3dfe2e00668bce25dbd3dc8830df902ed2c7841ef1bc6e8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-common1-abbc59b.stdout",
    "stdout_hash": "1e13ef807ef595ba00dffaece811bd41b66c4c73b9dcb6271cea0fdd",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}