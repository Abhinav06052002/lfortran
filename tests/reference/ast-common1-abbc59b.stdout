(TranslationUnit
    [(Program
        common1
        (TriviaNode
            []
            [(EndOfLine)
            (EndOfLine)]
        )
        []
        [(Implicit
            [(ImplicitSpec
                (AttrType
                    TypeReal
                    []
                    ()
                    ()
                    None
                )
                [(LetterSpec
                    a
                    z
                )]
            )]
            ()
        )]
        [(Declaration
            (AttrType
                TypeReal
                []
                ()
                ()
                None
            )
            []
            [(A
            []
            []
            ()
            ()
            None
            ())
            (B
            []
            []
            ()
            ()
            None
            ())]
            ()
        )
        (Declaration
            ()
            [(SimpleAttribute
                AttrCommon
            )]
            [(sample
            []
            []
            ()
            ()
            Slash
            ())
            (A
            []
            []
            ()
            A
            None
            ())
            (B
            []
            []
            ()
            B
            None
            ())]
            ()
        )
        (Declaration
            ()
            [(SimpleAttribute
                AttrCommon
            )]
            [(c
            []
            []
            ()
            ()
            Slash
            ())
            (c
            []
            []
            ()
            c
            None
            ())
            (b
            []
            []
            ()
            ()
            Slash
            ())
            (d
            []
            []
            ()
            d
            None
            ())
            (e
            []
            []
            ()
            e
            None
            ())
            (f
            []
            []
            ()
            f
            None
            ())
            (c
            []
            []
            ()
            ()
            Slash
            ())
            (g
            []
            []
            ()
            g
            None
            ())
            (h
            []
            []
            ()
            h
            None
            ())
            (i
            [(1
            5
            DimensionExpr)]
            []
            ()
            (FuncCallOrArray
                i
                []
                [(()
                5
                ()
                0)]
                []
                []
                []
            )
            None
            ())]
            ()
        )
        (Declaration
            ()
            [(SimpleAttribute
                AttrCommon
            )]
            [(()
            []
            []
            ()
            ()
            Slash
            ())
            (x
            []
            []
            ()
            x
            None
            ())
            (y
            []
            []
            ()
            y
            None
            ())
            (z
            [(1
            10
            DimensionExpr)]
            []
            ()
            (FuncCallOrArray
                z
                []
                [(()
                10
                ()
                0)]
                []
                []
                []
            )
            None
            ())]
            ()
        )]
        [(Assignment
            0
            A
            10
            ()
        )
        (Assignment
            0
            B
            20
            ()
        )
        (SubroutineCall
            0
            pass
            []
            []
            []
            []
            ()
        )
        (Stop
            0
            ()
            ()
            ()
        )]
        []
    )
    (Subroutine
        pass
        []
        []
        ()
        ()
        []
        []
        []
        [(Declaration
            (AttrType
                TypeReal
                []
                ()
                ()
                None
            )
            []
            [(A
            []
            []
            ()
            ()
            None
            ())
            (B
            []
            []
            ()
            ()
            None
            ())]
            ()
        )
        (Declaration
            ()
            [(SimpleAttribute
                AttrCommon
            )]
            [(sample
            []
            []
            ()
            ()
            Slash
            ())
            (A
            []
            []
            ()
            A
            None
            ())
            (B
            []
            []
            ()
            B
            None
            ())]
            ()
        )]
        [(Print
            0
            ()
            [A
            B]
            ()
        )
        (Return
            0
            ()
            ()
        )]
        []
        []
    )]
)
